src/AppInstallerCommonCore/NameNormalization.cpp

```diff
@@ -376,11 +378,14 @@ namespace AppInstaller::Utility
                 // Repeatedly remove matches for the regexes to create the minimum name
                 while (RemoveAll(ProgramNameRegexes, result.Name));
 
-                auto tokens = Split(ProgramNameSplit, result.Name, LegalEntitySuffixes);
-                result.Name = Join(tokens);
+                if (!PreserveWhiteSpace)

```

Comment:
Wow. I have no idea how I missread the code this bad; I assumed the split was around whitespace. My bad.
I believe the only thing that needs to be conditional is the `Remove(NonLettersAndDigits, ...)` because it also removes whitespace. I'll see if I can make a version like `NonLettersDigitsOrWhitespace`.